src/pages/register/Register.js

```diff
@@ -0,0 +1,98 @@
+import React, { useState } from "react";
+
+function Register() {
+  const [newEmail, setNewEmail] = useState("");
+  const [newUsername, setNewUsername] = useState("");
+  const [newPassword, setNewPassword] = useState("");
+  const [error, setError] = useState(null);
+
+  async function handleRegistration(e) {
+    e.preventDefault();
+    try {
+      const response = await fetch(
+        `https://localhost:8000/api/v1/auth/register`,
+        {
+          method: "POST",
+          headers: {
+            "Content-Type": "application/json",
+          },
+          body: JSON.stringify({
+            username: newUsername,
+            password: newPassword,
+            email: newEmail,
+          }),
+        },
+      );
+
+      const data = await response.json();
+
+      if (data.error | (data.status !== 200)) {
+        throw new Error(`HTTP error! Status: ${response.status}`);
+      }
+
+      //TODO: Logic for successful registration goes below here
+      window.alert(`Welcome, ${data.username}`);
+      setError(null);
+
+      return data;
+    } catch (error) {
+      console.error("Error during registration:", error);
+      setError("Error during registration. Please try again.");
+    }
+  }
+
+  return (
+    <div className="registerPage">
+      <h1>Register</h1>
+
+      <form
+        onSubmit={(e) => handleRegistration(e)}
+        className="registerContainer"
+      >
+        <div>
```

Comment:
Inputs should be associated with a label. Instead of wrapping each input in a div, wrap them in a label like this:
```
<label>Email:
    <input type="email" (...other attributes) />
</label>
```
it is also ok to do it this way:
``` 
<label htmlFor="myEmailInput" >Email:</label>
<input id="myEmailInput" type="email" (...other attributes) />
```

In the second way the htmlFor and id attributes have to have the same name. This is how the two are recognized as belonging together under the hood and by screen readers. When you wrap the label around the input (like the first way) they are implicitly associated with each other so you can leave the htmlFor and id attributes off.